extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java

```diff
@@ -675,8 +675,24 @@ public static List<DecoratorBuildItem> createInitJobDecorators(String target, St
                 .map(Optional::get)
                 .collect(Collectors.toList());
 
+        List<AddImagePullSecretDecorator> imagePullSecretDecorators = decorators.stream()
+                .filter(d -> d.getGroup() == null || d.getGroup().equals(target))
+                .map(d -> d.getDecorator(AddImagePullSecretDecorator.class))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+
         items.stream().filter(item -> item.getTarget() == null || item.getTarget().equals(target)).forEach(item -> {
 
+            for (final AddImagePullSecretDecorator delegate : imagePullSecretDecorators) {
+                result.add(new DecoratorBuildItem(target, new NamedResourceDecorator<PodSpecBuilder>("Job", item.getName()) {
+                    @Override
+                    public void andThenVisit(PodSpecBuilder builder, ObjectMeta meta) {
+                        delegate.andThenVisit(builder, meta);
+                    }
+                }));
+            }
+
```

Comment:
Why do we need this? The `AddImagePullSecretDecorator` decorators are executed before the decorator that creates the Job is executed and then it's not adding the image pull secret to the pods of the Job resource?

Reply:
We need that because we want to replay the `AddImagePullSecretDecorator` on the Job.

Regarding the order of things,  all `ResourceProvidingDecorator` including the `AddJobResourceDecorator` are meant to be executed before any `NamedResourceDecorator` (which AFAIR is the reason `ResourceProvidingDecorator` exists in the first place) so I don't see a problem with that (btw the included test verifies that).

Reply:
> We need that because we want to replay the `AddImagePullSecretDecorator` on the Job.
> 

I see. The `AddImagePullSecretDecorator` decorators that are registered in `KubernetesCommonHelper.createPodDecorators` will only apply to resources with a specific name. 

I think the solution here is a bit hacky and the right place where this logic should go is in `InitTaskProcessor.process` because is where the Job resource decorators are registered and do something like:

```
config.getImagePullSecrets().ifPresent(l -> {
    l.forEach(s -> result.add(new DecoratorBuildItem(target, new AddImagePullSecretDecorator(jobName, s))));
});
```

Reply:
It doesn't really matter what the original decorator is doing and where it gets applied, as we direclty delegate to the `andThenVisit` that implements the decoration logic. All the filtering logic is implemented by the `visit` method which we bypass.

What you suggest is not enough as we have pull secrets added by other sources (e.g. container image extensions) and not just user configuration.

The approach followed by this pull request follows what we already do for env vars, volumes and mounts of Init Jobs. 
We can discuss about ways of improving this in another pull request.